src/docs/arch-connection.dox

```diff
@@ -0,0 +1,56 @@
+/*! @arch_page arch-connection Connection
+
+@section arch_conn_def Definition
+
+A connection is a handle to a WiredTiger database instance. The connection has exclusive access to
+the database through a file lock, hence only one connection can be opened at a time. Internally, a
+connection is represented by \c WT_CONNECTION.
+
+@section arch_conn_lifecycle Life cycle
+
+@subsection arch_conn_init Initialization
+
+A connection is initialized when WT_CONNECTION::wiredtiger_open is called by the user application.
+WT_CONNECTION::wiredtiger_open accepts a list of configuration items (see @ref database_config) that
+can be used to enable different WiredTiger features and tune their behavior. Those features are for
+example related to @ref arch-eviction, @ref arch-logging, @ref arch-checkpoint, @ref arch-cache,
+statistics, etc. All the different available configuration settings are described in the
+documentation for WT_CONNECTION::wiredtiger_open.
+
+WT_CONNECTION::wiredtiger_open also performs different sanity checks depending on the configuration
+item "create". When "create" is specified and a database does not already exist, a new database is
+created as well as specific WiredTiger files such as the turtle file and other metadata files. If a
+database already exists, WiredTiger will try to open it and check for the existence of the different
+required WiredTiger files. If "create" is not specified, WiredTiger expects a database where it is
+executed. In case of a failure during WT_CONNECTION::wiredtiger_open, the connection is not created
+and the returned error can be either \c WT_RUN_RECOVERY or \c WT_TRY_SALVAGE (see @ref
```

Comment:
and the returned error can be either \c WT_RUN_RECOVERY or \c WT_TRY_SALVAGE (see @ref error_handling) ->
and either \c WT_RUN_RECOVERY or \c WT_TRY_SALVAGE error is retuned to the user application. In case API caller receives either of these errors, a recovery operation will be required to bring database to a consistent state (see @ref command_line for more details).

Reply:
61302c484cfc4d30b941b2c8b84414259ff5d4d3